guides/manual/1-overview.md

```diff
@@ -1,71 +1,151 @@
+      Hello {{result.data.name}}!
+    </:content>
+
+    <:loading as |state|>
+      <Spinner @progress={{state.completedRatio}} />
+      <button {{on "click" state.abort}}>Cancel?</button>
+    </:loading>
+  </Request>
+</template>
+```
+
+Web clients are like high-latency, remotely distributed, often-stale partial replicas of server state - but requests using WarpDrive utilize an [advanced relational cache](./5-caching.md) that simplifies these problems, solving them when it can and providing intelligent escape valves for when it can't so that no matter what you can quickly get the data you need in the right state.
+
+WarpDrive's reactive objects transform raw cached data into rich, reactive data. The resulting objects are immutable, always displaying the latest state in the cache while preventing accidental or unsafe mutation in your app. The output and [transformation](./concepts/transformation.md) is controlled by a simple JSON [ResourceSchema](./concepts/schemas.md).
+
+```ts
+import { withDefaults } from '@warp-drive/schema-record';
+
+store.schema.registerResource(
+  withDefaults({
+    type: 'user',
+    fields: [
+      { kind: 'field', name: 'firstName' },
+      { kind: 'field', name: 'lastName' },
+      { kind: 'field', name: 'birthday', type: 'luxon-date' },
+      {
+        kind: 'derived',
+        name: 'age',
+        type: 'duration',
+        options: { field: 'birthday', format: 'y' }
+      },
+      {
+        kind: 'derived',
+        name: 'fullName',
+        type: 'concat',
+        options: { fields: ['firstName', 'lastName'], separator: ' ' }
+      },
+    ]
+  })
+)
+```
+
+[Mutation](./concepts/mutations.md) is handled within controlled contexts. The data to edit is "checked out" for editing, giving access to a mutable version. Local edits are seamlessly preserved if the user navigates away and returns without saving, and the changes are buffered from appearing elsewhere in your app until they are also committed to the server.
+
+```ts
+import { Checkout } from '@warp-drive/schema-record';
+
+// ...
+
+const editable = await user[Checkout]();
+editable.firstName = 'Chris';
+```
+
+WarpDrive is only semi-opinionated about your API. Almost every API is compatible just by authoring a [request handler](./concepts/handlers.md) to ensure that the responses are normalized into the cache format.
+
+```ts
+const NormalizeKeysHandler = {
+  request(context, next) {
+    return next(context.request).then((result) => {
+      return convertUnderscoredKeysToCamelCase(result.content);
+    });
+  }
+}
+```
+
+The [cache](https://github.com/emberjs/data/blob/main/packages/core-types/src/cache.ts) itself is also pluggable if using a different cache format would work better for the demands of your API.
+
+```ts
+import Store from '@ember-data/store';
+import { AppCache } from './my-cache';
+
+class AppStore extends Store {
+  createCache(capabilities) {
+    return new AppCache(capabilities);
+  }
+}
+```
```

Comment:
```suggestion
```ts
import Store from '@ember-data/store';
import { CustomCache } from './my-cache';

class AppStore extends Store {
  createCache(capabilities) {
    return new CustomCache(capabilities);
  }
}
```
```

Reply:
added directly since this had a formatting error